src/lib/control/v1/kms.js

```diff
@@ -2,59 +2,76 @@ import {encrypt} from '../../kms';
 
 export default async (req, res, next) => {
   const plaintext = req.body.secret;
-  let keyObj;
+  const submittedKeys = (
+    (req.body.keys instanceof Array) ?
+      req.body.keys :
+      [req.body.keys]
+  ).filter((k) => !!k);
 
-  try {
-    keyObj = JSON.parse(req.body.key);
-  } catch (err) {
-    console.log(err);
-    keyObj = null;
-  }
-
-  if (!keyObj) {
-    return next(new Error('CMK not defined'));
-  }
-
-  if (!keyObj.hasOwnProperty('region') || !keyObj.hasOwnProperty('key')) {
-    return next(new Error('Invalid key format'));
+  if (!submittedKeys || submittedKeys.length === 0) {
+    return next(new Error('KMS key(s) not selected'));
   }
 
   if (!plaintext) {
     return next(new Error('Plaintext not entered'));
   }
 
-  const region = keyObj.region;
-  const key = keyObj.key;
-  const params = {
-    KeyId: key,
-    Plaintext: plaintext,
-    region
-  };
+  // Deserialize form data
+  const keys = submittedKeys.map((body) => {
+    const invalidKeyError = new Error('Invalid key format');
+    let parsedBody;
+
+    try {
+      parsedBody = JSON.parse(body);
+    } catch (ex) {
+      return invalidKeyError;
+    }
+    const {region, account, key} = parsedBody;
+
+    console.log(parsedBody);
```

Comment:
Do we want some additional context around this logging statement?

Reply:
Derp, left in some debugging code :-|